code/modules/organs/subtypes/parasite.dm

```diff
@@ -215,17 +215,53 @@
 	icon_state = "blacktumor"
 	dead_icon = "blacktumor"
 
-	organ_tag = "zombie"
-
-	parent_organ = BP_CHEST
+	organ_tag = BP_ZOMBIE_PARASITE
+	parent_organ = BP_HEAD
 	stage_interval = 150
 
+	var/last_heal = 0
+	var/heal_rate = 5 SECONDS
+
 /obj/item/organ/internal/parasite/zombie/process()
 	..()
 
 	if (!owner)
 		return
 
+	if(length(owner.bad_external_organs) && last_heal + heal_rate < world.time && isundead(owner))
+		var/list/organs_to_heal = owner.bad_external_organs
+		shuffle(organs_to_heal)
+		for(var/thing in organs_to_heal)
+			var/obj/item/organ/external/O = thing
+			if(istype(O, /obj/item/organ/external/head)) // the head is the weak point
```

Comment:
Is this true for all undead in `isundead` or just zombies? If not, you can use the `iszombie` thing you just defined.